js/classes/entities/midground/chad.js

```diff
@@ -654,9 +690,14 @@ class Chad {
         // Step 6: Check for any zone-specific conditions
         if (ZONE.name === "River Start") {
             if (this.pos.y > Vector.blockToWorldSpace(new Vector(0, 28)).y) {
-                this.takeDamage(Chad.MAX_HEALTH);
+                this.takeDamage(this.maxHealth);
             }
         }
+
+        // Step 7: Check general zone conditions
+        // if (this.pos.y > Vector.blockToWorldSpace(new Vector(0, 75)).y) {
+        //     this.takeDamage(this.maxHealth);
+        // }
```

Comment:
Is this needed?

Reply:
I wrote that to solve the problem of Chad not dying when he falls into the bottomless pit on the right side in the endzone. I ended up commenting it out because certain zones were lower altitude, causing Chad to instantly die when entering them. I'll either get rid of it or make it functional.

Reply:
Nice! We could make it so that he dies when passing a distance that is further in the y direction than he can go on any map perhaps. It would maybe take a while for him to reach that distance but might be a good option. Alternatively you could just add a border at the bottom of the end zone so that he just falls into it and teleports to the village.